simulation.cpp

```diff
@@ -169,8 +169,14 @@ void tick(simulation &s)
         switch_target(s.get_env());
     }
 
+    if(s.get_pop().get_inds().size()){
```

Comment:
is this if statement here because you want to check if there is a population in the first place?
As a rule of thumb, if you see more than 1 dots in your call, you should wrap up the function call into  one function. This is called Demeter law, if you respect it the code is gonna be much nicer and readable